src/components/KSelect/KSelect.vue

```diff
@@ -713,9 +716,12 @@ watch(() => props.items, (newValue, oldValue) => {
   }
 }, { deep: true, immediate: true })
 
-watch(filterQuery, (q: string) => {
-  emit('query-change', q)
-})
+watch(filterQuery, (query: string) => {
+  if (!skipQueryChangeEmit.value) {
+    emit('query-change', query)
+  }
+  skipQueryChangeEmit.value = false
+}, { immediate: true })
```

Comment:
Yes, that is more of backwards compatibility thing. In this PR I'm removing an emit on line 580 that used to fire every time input is in focus. What it was doing previously is fire this event with empty string (since we're resetting it) pretty much every time the dropdown opens. The component test caught this. That's why I added `immediate: true` to still have that "initial" emit. I was doubting between keeping it and removing it but then I thought knowing the initial value right away doesn't hurt?

Reply:
isn't an initial emit signaling that the query changed even though it didn't?

Reply:
Well yeah I guess. I remove the `immediate: true` now but that created a problem where the component wasn't emitting when the value is refreshed on dropdown open so I had to tweak the condition in this watcher. Please re-review.